source/fuzz/transformation_equation_instruction.cpp

```diff
@@ -64,61 +65,93 @@ bool TransformationEquationInstruction::IsApplicable(
     }
   }
 
-  return MaybeGetResultType(ir_context) != 0;
-}
-
-void TransformationEquationInstruction::Apply(
-    opt::IRContext* ir_context,
-    TransformationContext* transformation_context) const {
-  fuzzerutil::UpdateModuleIdBound(ir_context, message_.fresh_id());
+  // Check that the module remains valid if insert new instruction with
+  // |opcode|.
+  auto opcode = static_cast<SpvOp>(message_.opcode());
+  switch (opcode) {
+    case SpvOpConvertUToF:
+    case SpvOpConvertSToF: {
+      if (message_.in_operand_id_size() != 1) {
+        return false;
+      }
 
-  opt::Instruction::OperandList in_operands;
-  std::vector<uint32_t> rhs_id;
-  for (auto id : message_.in_operand_id()) {
-    in_operands.push_back({SPV_OPERAND_TYPE_ID, {id}});
-    rhs_id.push_back(id);
-  }
+      const auto* type = ir_context->get_type_mgr()->GetType(
+          fuzzerutil::GetTypeId(ir_context, message_.in_operand_id(0)));
+      if (!type) {
+        return false;
+      }
 
-  FindInstruction(message_.instruction_to_insert_before(), ir_context)
-      ->InsertBefore(MakeUnique<opt::Instruction>(
-          ir_context, static_cast<SpvOp>(message_.opcode()),
-          MaybeGetResultType(ir_context), message_.fresh_id(), in_operands));
+      if (const auto* vector = type->AsVector()) {
+        // Check that the operand has valid type.
+        if (!vector->element_type()->AsInteger() ||
+            vector->element_type()->AsInteger()->IsSigned() !=
```

Comment:
I am not sure that the sign needs to correspond to the instruction, actually.

SPIR-V is really relaxed about sign.

Please check whether it's in fact legal to apply OpConvertSToF to an unsigned scalar/vector.  If it is then it would be better to let the transformation (and fuzzer pass) be more relaxed here.

Reply:
Fixed in a6ad6b3.